### PR TITLE
sherlock: unstable-2023-10-06 -> unstable-2024-05-12

### PR DESCRIPTION
**Pull request by AwesomeQubic**
## Description of changes

Updated sherlock to latest git commit (note I do not think sherlock publishes a change log)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by JohnRTitor — COMMENTED** [review]
Welcome to Nixpkgs and thanks for your contribution.

Could you also add `unstableGitUpdater` update script, like https://github.com/NixOS/nixpkgs/pull/310859/commits/1ca55c244030546dc1e2ce4df762a58f5164aefd?

This will let bots like @r-ryantm automatically bump these.

**Review by JohnRTitor — CHANGES_REQUESTED** [review]
General conventions for update commit names like this:
```
package-name: prev-version -> new-version
```
Please update the commit message, using `git commit --amend`
Then force push the changes using `git push --force-with-lease`

**Comment by AwesomeQubic** [comment]
I can't find documentation for `unstableGitUpdater`. Could you point me to place where its documented/anything I can refer myself to while implementing it?

**Comment by JohnRTitor** [comment]
It is called [unstable-updater.nix](https://github.com/NixOS/nixpkgs/blob/5a1112cb087b1d2ebec6a33657cb6aca12be3bc1/pkgs/common-updater/unstable-updater.nix). It's not well documented, you see.. but it has to be used like [this](https://github.com/NixOS/nixpkgs/pull/310859/commits/1ca55c244030546dc1e2ce4df762a58f5164aefd).

To run the update script, or any update script in nixpkgs for that matter.
```bash
nix-shell maintainers/scripts/update.nix --argstr package hello-world
```

**Review by JohnRTitor — APPROVED** [review]
Result of `nixpkgs-review pr 311087` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sherlock</li>
  </ul>
</details>

**Comment by AwesomeQubic** [comment]
like this?

File: pkgs/tools/security/sherlock/default.nix [hunk 1]
```diff
@@ -2,18 +2,19 @@
...
 
 python3.pkgs.buildPythonApplication rec {
   pname = "sherlock";
-  version = "unstable-2023-10-06";
+  version = "unstable-2024-05-12";
   format = "other";
 
   src = fetchFromGitHub {
     owner = "sherlock-project";
     repo = pname;
```
Comment (JohnRTitor) [inline pkgs/tools/security/sherlock/default.nix:15]:
```suggestion
    repo = "sherlock";
```
Being explicit is better here. Else if for some reason, pname changes, it'll break the whole derivation.

**Review by JohnRTitor — CHANGES_REQUESTED** [review]

**Comment by JohnRTitor** [comment]
> like this? (also I'm not sure it works, so I would advice double checking it)

Yes I checked this branch out and tested. To test this yourself, you can set `version`, `rev`, and `hash` to garbage value. And run 
```bash
nix-shell maintainers/scripts/update.nix --argstr package sherlock
```

It would not change anything from your changes though. Since you are already on a latest commit.
The version will be prepended with `0-`, that is fine, according to the convention (https://github.com/NixOS/nixpkgs/pull/234201). Commit that.

**Comment by AwesomeQubic** [comment]
as per your suggestion I have set pname to an explicit value

**Comment by JohnRTitor** [comment]
Nice. As the last step, the last two of your commits can be squashed into your first commit.
Use `git rebase --interactive HEAD~3`, change the last two commits from `pick` to `squash`. Save changes. 
Then force push.

**Comment by AwesomeQubic** [comment]
Done, also thanks for your help.

**Review by JohnRTitor — APPROVED** [review]
Builds and runs successfully.

I hope this will be merged soon.

Thanks @AwesomeQubic for being so patient and responsive in your first PR to Nixpkgs. Wish you a well journey ahead.

**Comment by AwesomeQubic** [comment]
> Thanks @AwesomeQubic for being so patient and responsive in your first PR to Nixpkgs. Wish you a well journey ahead.

I also would like to thank you for giving me guidance.

Now I guess we will just have to wait :>

**Review by isabelroses — APPROVED** [review]

**Pull request merged**